tests/integration/targets/provision_virtual_esxi_test/runme.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+source ../init.sh
+ansible-playbook mock_side_effects.yml  &
```

Comment:
What about having an async task doing the side effect, instead of another ansible-playbook execution?
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_async.html